docs/handling-reports.md

```diff
@@ -13,27 +13,19 @@
 4. Now comes the fun part, discussing the vulnerability! (assuming it needs more
    investigation and hasn't been ruled out).
 
-Discussing the vulnerability can range from being extremely sensitive to "fine
-to try to keep it private for now". There's a few venues that the response can
-be discussed at:
-
-* First, you can use `security@rust-lang.org`. This may or may not include the
-  original reporter. Keep in mind that email clients have varying degrees of
-  caching and such, super sensitive stuff should always be encrypted. You get to
-  always choose who sees what here, but it's not a great medium to develop with
-  or make public later.
-
-* Next you can use the `#security` channel in Discord. Keep in mind the core
-  team can see messages here as well. Users (like the reporter) cannot be added
-  here.
-
-* We've also started to test out using GitHub Advisories. This is on the
-  "security" tab on GitHub. Keep in mind the core team can see all messages here
-  as well. You can add anyone to the conversation here on the bug (including the
-  reporter).
-
-Over the next few days you'll develop a response to the security issue in these
-venues, and it's recommended to do this in collaboration with the original
+The preferred venue for discussing the vulnerability and plan a fix is a Zulip
+stream (on [the rust-lang instance](https://rust-lang.zulipchat.com)) dedicated
+to the vulnerability. A dedicated stream allows us to grant access to the
+reporter or other team members without letting them access other discussions.
+
+The stream should be named `wg-security-response/<date-of-the-report>`, it
+should be private and it should allow new members to read past messages. Keep
+in mind that, while organization admins can't read the chat contents even if
+they have full access to the Zulip instance, they can read the stream name:
+avoid putting hints of what's going on in the name or the description.
```

Comment:
How do we deal with this stream after the incident is resolved? Can we close/archive it?

Personally, I don't want my Zulip sidebar cluttered with historical streams, but if I just unsubscribe would I lose access? Even if we close it somehow, I may still want to be _able_ to refer back to it later.

Reply:
Yeah I think unsubscribing will cause you to lose access.

A solution could be to have a policy to make the stream public after the fix is deployed. That way we increase the transparency and we can re-subscribe when needed.

Reply:
So, zulip in general doesn't handle archiving streams very well..

Reply:
..one thing we can do is to rename it to something like `zz-foo`. You can also delete the stream, in which case the messages still exist but can only be found by linking.

Reply:
Making it public is interesting. I wonder how that interacts, side note, with zulip-archive.rust-lang.org

Reply:
One final note is that, for me at least, zulip does a good job of moving streams that are not in active use into some hidden view where they don't bother me (though occasionally this confuses me because I can't find them)

Reply:
I'm not keen on making it public, actually, because participants may have shared some details with the understanding that it was private. Especially if we're ever coordinating with third parties for a patch release, for instance.

Reply:
Since I wasn't really involved in the latest incident, I decided to just try unsubscribing. I can still find it in "all streams", and I can even view it, but I can't re-subscribe. I assume I wouldn't see any new messages either, just those that I saw while I was subscribed.

That seems OK. If there's a way to lock the stream (once resolved) so there will definitely be no more messages, then everyone could feel free to unsubscribe without missing anything.

Reply:
@nikomatsakis it seems you're describing "demotion" for inactive streams, and there's a setting if you don't actually want that: https://zulipchat.com/help/manage-inactive-streams